src/ctx.rs

```diff
@@ -0,0 +1,172 @@
+//! This module provided types for context representation could be used in context-sensitive parsing.
+//! See [ctx attribute](attributes/index.html#ctx) for more information.
+
+use core::ops::{Deref, DerefMut};
+use core::str::FromStr;
+
+/// An endian.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum Endian {
```

Comment:
Note: At some point we might add `Network` (which == big)
`deku(endian = "network")` for clarity

Reply:
Or, the `from_str` "network" -> `Endian::Big` might be better

Reply:
We can mark Endian `#[non_exhaustive]` then add a new variant called `Network` to `enum Endian` and treat it as `Big`.(or other way) I dislike the idea of "network" -> `Endian::Big`.

Reply:
It could be done in https://github.com/sharksforarms/deku/pull/61/files/5e7deecaeda49f31229b2d691aa0367791a3485d#diff-3227a35819c9da6bf1bf643fad7e99a8R186